velox/functions/prestosql/ArrayMinMaxBy.cpp

```diff
@@ -0,0 +1,179 @@
+    auto& decodedArray = *arrayDecoder.get();
+
+    // Flatten input array.
+    auto flatArray = flattenArray(rows, args[0], decodedArray);
+
+    std::vector<VectorPtr> lambdaArgs = {flatArray->elements()};
+    auto newNumElements = flatArray->elements()->size();
+
+    SelectivityVector validRowsInReusedResult =
+        toElementRows<ArrayVector>(newNumElements, rows, flatArray.get());
+
+    VectorPtr newElements;
+    auto elementToTopLevelRows = getElementToTopLevelRows(
+        newNumElements, rows, flatArray.get(), context.pool());
+
+    // Loop over lambda functions and apply these to elements of the base array.
+    // In most cases there will be only one function and the loop will run once.
+    auto it = args[1]->asUnchecked<FunctionVector>()->iterator(&rows);
+    while (auto entry = it.next()) {
+      auto elementRows = toElementRows<ArrayVector>(
+          newNumElements, *entry.rows, flatArray.get());
+      auto wrapCapture = toWrapCapture<ArrayVector>(
+          newNumElements, entry.callable, *entry.rows, flatArray);
+
+      entry.callable->apply(
+          elementRows,
+          &validRowsInReusedResult,
+          wrapCapture,
+          &context,
+          lambdaArgs,
+          elementToTopLevelRows,
+          &newElements);
+    }
+
+    // New elements after the lambda function is applied
+    exec::LocalDecodedVector decodedElements(
+        context, *newElements, validRowsInReusedResult);
+
+    BufferPtr resultIndices = allocateIndices(rows.size(), context.pool());
+    auto* rawResultIndices = resultIndices->asMutable<vector_size_t>();
+
+    BufferPtr resultNulls = allocateNulls(rows.size(), context.pool());
+    auto* mutableNulls = resultNulls->asMutable<uint64_t>();
+
+    auto indices = getMaxOrMinIndices(rows, *flatArray, *newElements, context);
+    rows.applyToSelected([&](auto row) {
+      auto offset = flatArray->offsetAt(row);
+      if (decodedElements->isNullAt(offset)) {
+        bits::setNull(mutableNulls, row);
+      }
+      rawResultIndices[row] = indices[row];
+    });
+
+    auto localResult = BaseVector::wrapInDictionary(
+        resultNulls, resultIndices, rows.end(), flatArray->elements());
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+ private:
+  static std::vector<int> getMaxOrMinIndices(
+      const SelectivityVector& rows,
+      const ArrayVector& inputArray,
+      const BaseVector& inputElements,
+      exec::EvalCtx& context) {
+    std::vector<int> resultindices;
+    const SelectivityVector inputElementRows =
+        toElementRows(inputElements.size(), rows, &inputArray);
+    exec::LocalDecodedVector decodedElements(
+        context, inputElements, inputElementRows);
+    const auto* baseElementsVector = decodedElements->base();
+
+    // Allocate new vectors for indices.
+    BufferPtr indices = allocateIndices(inputElements.size(), context.pool());
+    vector_size_t* rawIndices = indices->asMutable<vector_size_t>();
```

Comment:
It looks like you're only using this as a temporary so you can iterate over it. We already depend on boost you could use an irange instead:

```
      const auto range = boost::irange(offset, offset + size);
      int minMaxIndex =
          *std::max_element(range.begin(), range.end(), compareLogic);
      resultindices.push_back(minMaxIndex);
```

Reply:
* You'll have to update the lambda to take it's arguments by value instead of by reference too, but they're vector_size_t so no performance concerns there